okio-files/src/commonMain/kotlin/okio/Filesystem.kt

```diff
@@ -28,6 +28,16 @@ abstract class Filesystem {
   @Throws(IOException::class)
   abstract fun cwd(): Path
 
+  /**
+   * Returns the children of the directory identified by [dir], or null if [dir] is a not a
+   * directory.
+   *
+   * @throws IOException if [dir] cannot be read, such as if the current process doesn't have
+   *     access to [dir], or if there's a loop of symbolic links, or if any name is too long.
+   */
+  @Throws(IOException::class)
+  abstract fun list(dir: Path): List<Path>?
```

Comment:
Nullable list 😬

Reply:
My feeling was it’s more ergonomic than a try/catch if the filesystem changes out from under you. Thoughts?

Reply:
Oh I thought null was for when it was a file, but it's also for when it's not available? If the directory disappears I would expect it to have the same behavior as when a file disappears and you ask for its bytes.

Even for the case when it's a file I probably also want it to throw.

Reply:
I can do that. Perhaps with `listOrNul()` later if it becomes too annoying. (Follow-up ’cause I missed this when I merged.)

Reply:
https://github.com/square/okio/pull/795